compiler-core/src/package_interface.rs

```diff
@@ -0,0 +1,280 @@
+                        deprecation: deprecation.clone(),
+                        parameters: parameters.len(),
+                        constructors: if *opaque {
+                            vec![]
+                        } else {
+                            constructors
+                                .iter()
+                                .map(|constructor| constructor.name.clone())
+                                .collect()
+                        },
+                    },
+                );
+            }
+
+            // A public module constant.
+            Definition::ModuleConstant(ModuleConstant {
+                public: true,
+                name,
+                type_,
+                documentation,
+                supported_targets,
+                location: _,
+                annotation: _,
+                value: _,
+            }) => {
+                let _ = values.insert(
+                    name.clone(),
+                    ValueInterface {
+                        supported_targets: supported_targets.to_vec(),
+                        type_: TypeInterface::from_type(type_.as_ref()),
+                        deprecation: Deprecation::NotDeprecated,
+                        documentation: documentation.clone(),
+                    },
+                );
+            }
+
+            // A public top-level function.
+            Definition::Function(Function {
+                public: true,
+                name,
+                arguments,
+                deprecation,
+                return_type,
+                documentation,
+                supported_targets,
+                location: _,
+                end_position: _,
+                body: _,
+                return_annotation: _,
+                external_erlang: _,
+                external_javascript: _,
+            }) => {
+                let _ = values.insert(
+                    name.clone(),
+                    ValueInterface {
+                        supported_targets: supported_targets.to_vec(),
+                        deprecation: deprecation.clone(),
+                        documentation: documentation.clone(),
+                        type_: TypeInterface::Fn {
+                            parameters: arguments
+                                .iter()
+                                .map(|arg| TypeInterface::from_type(arg.type_.as_ref()))
+                                .collect(),
+                            return_: Box::new(TypeInterface::from_type(return_type)),
+                        },
+                    },
+                );
+            }
+
+            // Private definitions are not included.
+            Definition::Function(_) => {}
+            Definition::CustomType(_) => {}
+            Definition::ModuleConstant(_) => {}
+
+            // Type aliases are ignored since those are also erased from the
+            // generated documentation.
+            Definition::TypeAlias(_) => {}
```

Comment:
Not true, they can have the pub keyword!

Reply:
Ok, I'll include type aliases as well!